src/transformers/models/clvp/tokenization_clvp.py

```diff
@@ -0,0 +1,546 @@
+                is_split_into_words=is_split_into_words,
+                pad_to_multiple_of=pad_to_multiple_of,
+                return_tensors=return_tensors,
+                return_token_type_ids=return_token_type_ids,
+                return_attention_mask=return_attention_mask,
+                return_overflowing_tokens=return_overflowing_tokens,
+                return_special_tokens_mask=return_special_tokens_mask,
+                return_offsets_mapping=return_offsets_mapping,
+                return_length=return_length,
+                verbose=verbose,
+                **kwargs,
+            )
+
+            encodings["input_ids_with_special_tokens"] = encodings_with_special_tokens["input_ids"]
+            encodings["attention_mask_with_special_tokens"] = encodings_with_special_tokens["attention_mask"]
+
+        return encodings
+
+    def _decode(
+        self,
+        token_ids: List[int],
+        skip_special_tokens: bool = False,
+        clean_up_tokenization_spaces: bool = True,
+        spaces_between_special_tokens: bool = True,
+        **kwargs,
+    ) -> str:
+        self._decode_use_source_tokenizer = kwargs.pop("use_source_tokenizer", False)
+
+        filtered_tokens = self.convert_ids_to_tokens(token_ids, skip_special_tokens=skip_special_tokens)
+        legacy_added_tokens = set(self._added_tokens_encoder.keys()) - set(self.all_special_tokens) | {
+            token for token in self.additional_special_tokens if self.convert_tokens_to_ids(token) >= self.vocab_size
+        }
+        sub_texts = []
+        current_sub_text = []
+        for token in filtered_tokens:
+            if skip_special_tokens and token in self.all_special_ids:
+                continue
+            if token in legacy_added_tokens:
+                if current_sub_text:
+                    string = self.convert_tokens_to_string(current_sub_text)
+                    if len(string) > 0:
+                        sub_texts.append(string)
+                    current_sub_text = []
+                sub_texts.append(token)
+            else:
+                current_sub_text.append(token)
+
+        if current_sub_text:
+            sub_texts.append(self.convert_tokens_to_string(current_sub_text))
+
+        if spaces_between_special_tokens:
+            text = " ".join(sub_texts)
+        else:
+            text = "".join(sub_texts)
+
+        clean_up_tokenization_spaces = (
+            clean_up_tokenization_spaces
+            if clean_up_tokenization_spaces is not None
+            else self.clean_up_tokenization_spaces
+        )
+
+        if clean_up_tokenization_spaces:
+            clean_text = self.clean_up_tokenization(text)
+            return clean_text
+        else:
+            return text
+
+    def clean_up_tokenization(self, text):
+        text = "".join(text)
+        text = (
+            text.replace("[SPACE]", " ")
+            .replace("[STOP]", "")
+            .replace(self.unk_token, "")
```

Comment:
if these tokens are special they can be skipped using `skip_special_tokens` rather than removing them completely.

Reply:
These tokens are a part of vocab 
```
"vocab":{"[STOP]":0,"[UNK]":1,"[SPACE]":2,"!":3,"'":4,"(":5,")":6,",":7,"-":8,".":
```
I think we should check these tokens before removing, something similar to this maybe -
```python
text = text.replace("[SPACE]", " ") if "[SPACE]" in self.encoder else text
text = text.replace("[STOP]", " ") if "[STOP]" in self.encoder else text
```
WDYT?

Reply:
If you do something like `tokenizer.add_tokens(["[SPACE]","[STOP]", special_tokens = True)`, then they will 1. never be split if encountered, and 2, when decoding they will be skipped (as they will always be part of the vocab (if not part of the BPE vocab, will be added to the `self.added_tokens_decoder`

Reply:
So it should be `text = text.replace("[SPACE]", " ") if "[SPACE]" in list(self.encoder.keys()+self.added_tokens_decoder) else text` right?

Sorry if I am missing something here.